utils/archive.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package utils
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// UnzipToPath extracts a given zip archive into a given path.
+// It returns a list of extracted paths.
+func UnzipToPath(zipFile io.ReaderAt, size int64, outPath string) ([]string, error) {
+	rd, err := zip.NewReader(zipFile, size)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create reader: %w", err)
+	}
+
+	paths := make([]string, len(rd.File))
+	for i, f := range rd.File {
+		filePath := filepath.Clean(strings.ReplaceAll(f.Name, "..", ""))
```

Comment:
This turns `path/to/file/foo...bar` into `path/to/file/foo.bar`, which doesn't seem appropriate.

Reply:
It doesn't, but neither the input seems all that appropriate. I've decided to be dumb there as I wasn't sure it was worth it to deal with such special cases and merely focused on being safe. But if you prefer we can be a bit smarter and either do a validation check and error out or even accept such filenames.

Reply:
@jupenur 

I've updated this with a slightly smarter version that doesn't touch the filename since this could actually derive from user uploaded files so it makes sense to be more lenient on that front. Still not worrying about possible malformed directory names (e.g. `bad..dirname`) since those are really not expected.